.travis.yml

```diff
@@ -32,74 +32,81 @@ addons:
 
 matrix:
     include:
+         # Try MacOS X
+         - os: osx
+           env: JOB="Py2" PYTHON_VERSION=2.7 SETUP_CMD='test'
+                CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
+
+         # Try on Linux
          - os: linux
-           env: PANDAS_VERSION="=0.16.2"
+           env: JOB="Py2" PYTHON_VERSION=2.7 SETUP_CMD='test'
+                CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
          - os: linux
-           env: TEST_MODE='figure' MATPLOTLIB_VERSION="=1.5.1"
+           env: JOB="Py2PandasPrev" PYTHON_VERSION=2.7 SETUP_CMD='test'
+                PANDAS_VERSION=$PREVIOUS_PANDAS CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
          - os: linux
-           env: TEST_MODE='astropy-dev'
+           env: JOB="Py2AstropyDev" PYTHON_VERSION=2.7 SETUP_CMD='test'
+                ASTROPY_VERSION='development' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
          - os: linux
-           env: TEST_MODE='sphinx'
+           env: JOB="Py2Sphinx" PYTHON_VERSION=2.7 SETUP_CMD='build_sphinx -w'
+                CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
          - os: linux
-           env: TEST_MODE='online'
+           env: JOB="Py2Coverage" PYTHON_VERSION=2.7 SETUP_CMD='test --online --coverage --cov-report=html'
+                CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2 PIP_DEPENDENCIES='pytest-cov'
          - os: linux
-           env: TEST_MODE='doctest'
+           env: JOB="Py2Doctest" PYTHON_VERSION=2.7 SETUP_CMD='build_sphinx -b doctest' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2 # doctest will fail till new version of astropy-helpers
          - os: linux
-           env: PYTHON_VERSION=3.4 TEST_MODE='python3'
+           env: JOB="Py2NumpyPrev" PYTHON_VERSION=2.7 SETUP_CMD='test' NUMPY_VERSION=1.9.2 CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
+         - os: linux
+           env: JOB="Py2Figures" PYTHON_VERSION=2.7 SETUP_CMD='test --figure' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
+         - os: linux
+           env: JOB="Py3" PYTHON_VERSION=3.4 MAIN_CMD='continuous-integration/python3.sh' SETUP_CMD='' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES
 
-# If the online build fails, we don't want it to report the whole build broken.
+    # allow_failures has to repeat the environment from the matrix above to mark it as such
     allow_failures:
-        - env: TEST_MODE='online'
-        - env: TEST_MODE='doctest'
-
-# Install miniconda
-before_install:
-    - source continuous-integration/travis/install_$TRAVIS_OS_NAME.sh
+         - env: JOB="Py2Doctest" PYTHON_VERSION=2.7 SETUP_CMD='build_sphinx -b doctest' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2 # doctest will fail till new version of astropy-helpers
+         - env: JOB="Py2Coverage" PYTHON_VERSION=2.7 SETUP_CMD='test --online --coverage --cov-report=html' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2 PIP_DEPENDENCIES='pytest-cov'
+         - env: JOB="Py2Figures" PYTHON_VERSION=2.7 SETUP_CMD='test --figure' CONDA_DEPENDENCIES=$CONDA_ALL_DEPENDENCIES2
```

Comment:
If we freeze dependancies for the figures build we should remove it from the allowed_faliures.